COMPATIBILITY.md

```diff
@@ -0,0 +1,80 @@
+# Compatibility Guarantee
+## Introduction
+
+The release of InfluxDB version 1 is a major milestone in the
+development of the software lifecycle. InfluxDB 1 is a stable platform
+for the growth of programs and projects that utilize InfluxDB as a
+database.
+
+## Expectations
+
+Although we expect that the vast majority of users will maintain this
+compatibility over time, it is impossible to guarantee that no future
+change will break any program. This document is an attempt to set
+expectations for the compatibility of InfluxDB 1 in the future. There
+are a number of ways in which a client that works today may fail to do
+so after a future point release. They are unlikely but worth recording.
+
+* Security. A security issue in the specification or implementation may
+  come to light whose resolution requires breaking compatibility. We
+  reserve the right to address such security issues.
+* Unspecified behavior. The InfluxDB specification tries to be explicit
```

Comment:
The reason I don't want to specifically cite certain things is because I think it will make the document less clear, not more. What we're doing is saying, "Everything user facing will be compatible except for these certain circumstances we reserve for ourselves." There's a bunch more that is included that I don't think we want to have to exhaustively list out. There's the graphite and collectd interfaces too that shouldn't change.

The list of things when we acquire new features will grow and that is covered by already saying, "If we add a feature in a new release, it's covered by this."

I also think we'll forget to update it when we add new things which will make the confusion greater. I think it's better to place a general concept over what compatibility we guarantee rather than trying to do essentially a list of all features in the product (because all of them are under the compatibility guarantee).

Although while I say that, I think that specifically citing things like whether disk formats can be relied on would be a good thing to do.